Added test for using numba.jit with lambdify

xavierdurawa opened the pull request:
Added a test to check that numba.jit is working
properly with lambdified functions.

<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->
Addresses #20070


#### Brief description of what is fixed or changed
New test was added to ensure that numba.jit works
with Sympy's lambdified functions

#### Other comments


#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

xavierdurawa committed:
Added test for using numba.jit with lambdify

Added a test to check that numba.jit is working
properly with lambdify to address Issue #20070.

eric-wieser commented on sympy/external/tests/test_numba.py:
```diff
@@ -0,0 +1,29 @@
+    f3 = lambdify((*X, *Y), X+Y)
+    xs = (1,2,3,4,5,6)
+    assert all(chain.from_iterable(f3(*xs) == numba.njit(f3)(*xs)))
+
```
```suggestion
```

xavierdurawa replied:
As in add numba to .travis.yaml?

eric-wieser replied:
No, removing the blank line

asmeurer replied:
You also should add it to Travis.

xavierdurawa replied:
not sure if I added numba to Travis correctly.  If I didn't, a suggestion or point in the right direction for how to do that would be appreciated.  Thanks

asmeurer replied:
You did, but it looks like you also need to add 

```
# numba
'*numba*',
```

here https://github.com/sympy/sympy/blob/master/bin/test_travis.sh#L188. You can check in the Travis build, this file should be one of the ones tested in the TEST_OPT_DEPENDENCY builds.

asmeurer commented on sympy/external/tests/test_numba.py:
```diff
@@ -0,0 +1,29 @@
+    # Test that numba.jit works with lambdify
+    x = symbols('x')
+    f1 = lambdify(x, sin(x), 'numpy')
+    assert numba.jit(f1)(1) == f1(1)
```
Numba jitted functions don't get the exact same numeric result as NumPy in general. You may want to use `assert abs(numba.jit(f1)(1) - f1(1)) < 1e-15` instead.

xavierdurawa committed:
Added test for using numba.jit with lambdify

Added a test to check that numba.jit is working
properly with lambdified functions

xavierdurawa committed:
Added test for using numba.jit with lambdify

oscarbenjamin commented on sympy/external/tests/test_numba.py:
```diff
@@ -0,0 +1,29 @@
+    Y = Matrix(symbols('y:3')).T
+    f3 = lambdify((*X, *Y), X+Y)
+    xs = (1,2,3,4,5,6)
+    element_comp = abs(f3(*xs) - numba.njit(f3)(*xs) < 1e-15)
```
This is calling `abs` on the bool array. The `1e-15` shouldn't be inside the bracket.

xavierdurawa committed:
Correcting parathesis placement

czgdp1807 committed:
Merge branch 'master' into issue_20070

czgdp1807 commented:
I have restarted the tests. @oscarbenjamin @asmeurer Are there any more changes needed to be made for this PR to be merged?
